python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
+        default=None,
+    )
+    dpia_location: str | None = Field(description="", default=None)
+    data_sensitivity_level: SecurityClassification | None = Field(
+        description="", default=SecurityClassification.OFFICIAL
+    )
+
+
+# Needs refining
+class AccessInformation(BaseModel):
+    """
+    Any metadata about how to access a data entity.
+    The same data entity may be accessable via multiple means.
+    """
+
+    where_to_access_dataset: str = Field(
+        description="User-friendly description of where the data can be accessed",
+        default="",
+    )
+    source_dataset_name: str = Field(description="", default="")
+    s3_location: str | None = Field(
+        description="Location of the data in s3", default=None
+    )
+
+
+# Needs refining
+class DataSummary(BaseModel):
+    """
+    Summarised information derived from the actual data.
+    """
+
+    row_count: int | None = Field(
+        description="Row count when the metadata was last updated", default=None
+    )
+
+
+class CustomEntityProperties(BaseModel):
+    """Custom entity properties not part of DataHub's entity model"""
+
+    usage_restrictions: UsageRestrictions = Field(
+        description="Limitations on how the data may be used and accessed",
+        default_factory=UsageRestrictions,
+    )
+    access_information: AccessInformation = Field(
+        description="Metadata about how to access a data entity",
+        default_factory=AccessInformation,
+    )
+    data_summary: DataSummary = Field(
+        description="Summary of data stored in this table", default_factory=DataSummary
+    )
+
+
+class Entity(BaseModel):
+    """
+    Any searchable data entity that is present in the metadata graph, which
+    may be related to other entities.
+    Examples include platforms, databases, tables
+    """
+
+    id: str | None = Field("Unique identifier for the entity. Relates to Datahub's urn")
+    display_name: str | None = Field("Display name of the entity")
+    name: str = Field("Actual name of the entity in its source platform")
+    fully_qualified_name: str | None = Field(
+        "Fully qualified name of the entity in its source platform"
+    )
+    description: str = Field(
+        description="Detailed description about what functional area this entity is representing, what purpose it has"
+        " and business related information.",
+    )
+    relationships: dict[RelationshipType, list[EntityRef]] = Field(
+        default={},
+        description="References to related entities in the metadata graph, such as platform or parent entities",
+    )
+    domain: DomainRef = Field(description="The domain this entity belongs to.")
+    governance: Governance = Field(description="Information about governance")
+    tags: list[TagRef] = Field(
+        default_factory=list,
+        description="Additional tags to add.",
+    )
+    # Needs refining
+    last_updated: Optional[datetime] = Field(
```

Comment:
I suggest we adopt Datahub's conventions for timestamps to avoid confusion

- `created`: A timestamp documenting when the asset was created in the source Data Platform (not on DataHub)
- `last_modified` instead of `last_updated`: A timestamp documenting when the asset was last modified in the source Data Platform (not on DataHub)